### PR TITLE
Ci/GitHub actions

### PR DESCRIPTION
**Pull request by szymonlesisz**
- removed `Makefile` in favor of shell scripts
- `travis` replaced by github actions

### CODE REVIEW

**Comment by tsusanka** [comment]
Just a very small comment since I am here: I would prefer to keep the Makefile and run the scripts because Makefile is somewhat a standard. That means: `make` -> `scripts/build-lib.sh`, `make publish` -> `scripts/publish-lib.sh`.

**Review by tsusanka — APPROVED** [review]
Just a very small comment since I am here: I would prefer to keep the Makefile and run the scripts from there because Makefile is somewhat a standard. That means: make -> scripts/build-lib.sh, make publish -> scripts/publish-lib.sh. Just a suggestion.

**Comment by szymonlesisz** [comment]
actually `Makefile` was/is kind of legacy solution used by Karel from [some crazy js boilerplate](https://github.com/trezor/trezor-link/commit/00593b2d876c32895dca8c5b76cf84f356a21b42#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52)

`Makefile` is not a standard in javascript world since all scripts all called from `yarn` or `npm` anyway. (like `yarn build`)

**Comment by tsusanka** [comment]
Aah and I can see `yarn build` [leads](https://github.com/trezor/trezor-link/pull/36/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R9-R10) to the script, I've missed that. Great then :)!

File: .npmignore [hunk 1]
```diff
@@ -2,11 +2,11 @@
 src
 babel.config.js
 .eslintrc
+.eslintignore
```
Comment (goodhoko) [inline .npmignore:5]: Seems like this should be commited separately?

File: .npmignore [hunk 2]
```diff
@@ -2,11 +2,11 @@
...
 .flowconfig
 .gitignore
 .gitmodules
-.travis.yml
-Makefile
```
Comment (goodhoko) [inline .npmignore:9]: This deleted line probably belongs to the previous commit.

File: README.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # trezor-link
 
-[![Build Status](https://travis-ci.org/trezor/trezor-link.svg?branch=master)](https://travis-ci.org/trezor/trezor-link) [![gitter](https://badges.gitter.im/trezor/community.svg)](https://gitter.im/trezor/community)
+[![Build Status](https://github.com/trezor/trezor-link/actions/workflows/tests.yml/badge.svg)](https://github.com/trezor/trezor-link/actions/workflows/tests.yml)
+[![NPM](https://img.shields.io/npm/v/trezor-link.svg)](https://www.npmjs.org/package/trezor-link)
```
Comment (goodhoko) [inline README.md:4]: This should be probably commited separately.

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        node-version: [10.x, 12.x, 14.x, 15.x]
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: yarn install
```
Comment (goodhoko) [inline .github/workflows/tests.yml:23]: Any reason why we don't use the bare `yarn` command here as we do elsewhere? (eg. in the `publish-lib.sh` from the previous commit.)

**Review by goodhoko — APPROVED** [review]
I all-handedly support removal of make - it loweres the amount of tooling which is indeed a welcomed change. 👍 

I can't judge if replacing Travis by GitHub improves anything, but I have nothing against that.

Most of my remarks are about git hygiene.

File: package.json [hunk 5]
```diff
@@ -4,8 +4,10 @@
   "description": "Trezor Link for browser",
   "main": "lib/index.js",
   "scripts": {
-    "build": "make build",
-    "check": "make check"
+    "type-check": "flow check src",
+    "lint": "eslint src",
+    "build:lib": "./scripts/build-lib.sh",
+    "publish:lib": "./scripts/publish-lib.sh"
```
[the remark this answers is not in the document]
Comment (szymonlesisz) [reply package.json:10]: future consistency with suite scripts

File: README.md (hunk 3 above)
Comment (szymonlesisz) [reply README.md:4]: why? this is scoped with replacing travis badges with github action badges

Comment (goodhoko) [reply README.md:4]: Sorry, I should have been clearer about what I mean.

>  this is scoped with replacing travis badges with github action badges

The replacement of build status badge is.

Removal of gitter badge and addition of NPM badge isn't. This addition and removal probably have a different reason thean the Travis/GitHub replacement. Ideally this commit only replaces the "build status" badge and then there's another commit with a message in the sense of
```
Update badges in readme

We do not use Gitter anymore. Remove its badge.
Add badge NPM badge for quick navigation to NPM.
```

File: .eslintignore [hunk 6]
```diff
@@ -0,0 +1 @@
+config_proto_compiled.js
```
[the remark this answers is not in the document]
Comment (szymonlesisz) [reply .eslintignore:1]: unnecessary complications.
makefile, which was using eslint job is replaced by shell script which is also running a eslint job but from different location (project root)
therefore moving it this to a different commit (before or after) will [result a regression](https://github.com/trezor/hd-wallet/pull/61#discussion_r611653422) in CI :D

File: scripts/build-lib.sh [hunk 7]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+# copy source files
+rm -rf lib
+cp -r src/ lib
```
[the remark this answers is not in the document]
Comment (szymonlesisz) [reply scripts/build-lib.sh]: updated

File: scripts/build-lib.sh [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+# remove tests and non js files
+rm -r lib/flow-test
+find lib/ -type f ! -name '*.js' | xargs -I {} rm {}
```
[the remark this answers is not in the document]
Comment (szymonlesisz) [reply scripts/build-lib.sh]: perfection is the enemy of good, leaving as it is

File: scripts/build-lib.sh [hunk 9]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
```
[the remark this answers is not in the document]
Comment (szymonlesisz) [reply scripts/build-lib.sh:2]: done

File: scripts/publish-lib.sh [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+yarn
+yarn type-check
+yarn lint
+yarn build:lib
+
+# publish
+npm version $1
+npm publish
+git push
+git push --tags
```
[the remark this answers is not in the document]
Comment (szymonlesisz) [reply scripts/publish-lib.sh]: done

File: .npmignore (hunk 1 above)
Comment (szymonlesisz) [reply .npmignore:5]: moved to commit c82b3ffbba4fe08e700ea9ee75568c0895411351

File: .npmignore (hunk 2 above)
Comment (szymonlesisz) [reply .npmignore:9]: indeed, moved to c82b3ffbba4fe08e700ea9ee75568c0895411351

File: README.md (hunk 3 above)
Comment (szymonlesisz) [reply README.md:4]: restoring gitter badge, feel free to remove it in separated commit in the future

File: .github/workflows/tests.yml (hunk 4 above)
Comment (szymonlesisz) [reply .github/workflows/tests.yml]: no reason, updated

**Comment by szymonlesisz** [comment]
@goodhoko 
> I can't judge if replacing Travis by GitHub improves anything, but I have nothing against that.
Travis is no longer supported as a free CI, so either this or no CI at all...

File: scripts/build-lib.sh (hunk 9 above)
[the remark this answers is not in the document]
Comment (goodhoko) [reply scripts/build-lib.sh:2]: 🖤

File: scripts/publish-lib.sh (hunk 10 above)
[the remark this answers is not in the document]
Comment (goodhoko) [reply scripts/publish-lib.sh]: 🖤

File: scripts/build-lib.sh (hunk 7 above)
[the remark this answers is not in the document]
Comment (goodhoko) [reply scripts/build-lib.sh]: 🖤

File: .npmignore (hunk 2 above)
Comment (goodhoko) [reply .npmignore:9]: 🖤

File: .github/workflows/tests.yml (hunk 4 above)
Comment (goodhoko) [reply .github/workflows/tests.yml]: 🖤

File: .eslintignore (hunk 6 above)
[the remark this answers is not in the document]
Comment (goodhoko) [reply .eslintignore:1]: Aah , I see, this connection didn't appear to me. Resolving.

File: .npmignore (hunk 1 above)
Comment (goodhoko) [reply .npmignore:5]: Thanks!

File: package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (goodhoko) [reply package.json:10]: Allright. Resolving.

**Comment by goodhoko** [comment]
@szymonlesisz Thanks for addressing all my input. Let's merge this. .)

**Comment by goodhoko** [comment]
> Travis is no longer supported as a free CI

ACK. Thanks for info. It would be nice to mention this in the commit message.

**Pull request merged**